### PR TITLE
Modify hiveParser's expression rules to implement AND/OR priority

### PR DESCRIPTION
**Pull request by jiefei30**
Modify hiveParser's expression rules to implement AND/OR's priority and make the structure of expression more clearer.

example, SQL: `SELECT c1, c2 FROM t1 WHERE c1 > 1 AND c2 < 3;`

old ParseTree :

old ParseTree

new ParseTree :

new ParseTree

### CODE REVIEW

File: sql/hive/v2/IdentifiersParser.g4 [hunk 1]
```diff
@@ -290,12 +290,10 @@ precedenceUnaryOperator
...
-    : KW_NULL
-    | KW_TRUE
-    | KW_FALSE
-    | KW_NOT KW_NULL
-    | KW_NOT KW_TRUE
-    | KW_NOT KW_FALSE
+    : (KW_NOT)? KW_NULL
+    | (KW_NOT)? KW_TRUE
+    | (KW_NOT)? KW_FALSE
+    | (KW_NOT)? KW_DISTINCT KW_FROM
```
Comment (KvanTTT) [inline sql/hive/v2/IdentifiersParser.g4:296]: No need to use parentheses here. Also, the common can be factored:

```suggestion
    : KW_NOT? (KW_NULL | KW_TRUE | KW_FALSE | KW_DISTINCT | KW_FROM)
```

File: sql/hive/v2/IdentifiersParser.g4 [hunk 2]
```diff
@@ -388,40 +374,25 @@ expression
     | expression precedenceConcatenateOperator expression
     | expression precedenceAmpersandOperator expression
     | expression precedenceBitwiseOrOperator expression
-    | expression precedenceSimilarExpressionPart
+    | expression precedenceComparisonOperator expression
+    | expression (KW_NOT)? precedenceRegexpOperator expression
```
Comment (KvanTTT) [inline sql/hive/v2/IdentifiersParser.g4:378]:
```suggestion
    | expression KW_NOT? precedenceRegexpOperator expression
```

File: sql/hive/v2/IdentifiersParser.g4 [hunk 3]
```diff
@@ -388,40 +374,25 @@ expression
...
+    | expression (KW_NOT)? KW_LIKE KW_ANY expressionsInParenthesis
+    | expression (KW_NOT)? KW_LIKE KW_ALL expressionsInParenthesis
```
Comment (KvanTTT) [inline sql/hive/v2/IdentifiersParser.g4:380]:
```suggestion
    | expression KW_NOT? KW_LIKE (KW_ANY | KW_ALL) expressionsInParenthesis
```

File: sql/hive/v3/IdentifiersParser.g4 [hunk 4]
```diff
@@ -290,12 +290,10 @@ precedenceUnaryOperator
...
-    : KW_NULL
-    | KW_TRUE
-    | KW_FALSE
-    | KW_NOT KW_NULL
-    | KW_NOT KW_TRUE
-    | KW_NOT KW_FALSE
+    : (KW_NOT)? KW_NULL
+    | (KW_NOT)? KW_TRUE
+    | (KW_NOT)? KW_FALSE
+    | (KW_NOT)? KW_DISTINCT KW_FROM
```
Comment (KvanTTT) [inline sql/hive/v3/IdentifiersParser.g4:296]:
```suggestion
    : KW_NOT? (KW_NULL | KW_TRUE | KW_FALSE | KW_DISTINCT KW_FROM)
```

File: sql/hive/v3/IdentifiersParser.g4 [hunk 5]
```diff
@@ -388,40 +374,25 @@ expression
     | expression precedenceConcatenateOperator expression
     | expression precedenceAmpersandOperator expression
     | expression precedenceBitwiseOrOperator expression
-    | expression precedenceSimilarExpressionPart
+    | expression precedenceComparisonOperator expression
+    | expression (KW_NOT)? precedenceRegexpOperator expression
```
Comment (KvanTTT) [inline sql/hive/v3/IdentifiersParser.g4:378]:
```suggestion
    | expression KW_NOT? precedenceRegexpOperator expression
```

File: sql/hive/v3/IdentifiersParser.g4 [hunk 6]
```diff
@@ -388,40 +374,25 @@ expression
...
+    | expression (KW_NOT)? KW_LIKE KW_ANY expressionsInParenthesis
+    | expression (KW_NOT)? KW_LIKE KW_ALL expressionsInParenthesis
+    | expression (KW_NOT)? KW_IN precedenceSimilarExpressionIn
+    | expression (KW_NOT)? KW_BETWEEN expression KW_AND expression
```
Comment (KvanTTT) [inline sql/hive/v3/IdentifiersParser.g4:382]:
```suggestion
    | expression KW_NOT? KW_IN precedenceSimilarExpressionIn
    | expression KW_NOT? KW_BETWEEN expression KW_AND expression
```

File: sql/hive/v2/IdentifiersParser.g4 (hunk 1 above)
Comment (jiefei30) [reply sql/hive/v2/IdentifiersParser.g4:296]: ok. got it

**Comment by KvanTTT** [comment]
Thank you for the contribution and welcome to grammars-v4 repository!

**Pull request merged**